File: frontend/src/methods/features.ts
// Copyright (c) 2026 Sidero Labs, Inc.
//
// Use of this software is governed by the Business Source License
// included in the LICENSE file.
import { createSharedComposable } from '@vueuse/core'
import { computed } from 'vue'

import { Runtime } from '@/api/common/omni.pb'
import type { FeaturesConfigSpec } from '@/api/omni/specs/omni.pb'
import { DefaultNamespace, FeaturesConfigID, FeaturesConfigType } from '@/api/resources'
import { useResourceWatch } from '@/methods/useResourceWatch'

export const useFeatures = createSharedComposable(() =>
  useResourceWatch<FeaturesConfigSpec>({
    resource: {
      type: FeaturesConfigType,
      namespace: DefaultNamespace,
      id: FeaturesConfigID,
    },
    runtime: Runtime.Omni,
  }),
)

export function useIsEnterprise() {
  const { data } = useFeatures()

  return computed(() => data.value?.spec.is_enterprise_image_factory ?? false)
}

File: internal/pkg/config/features.go
// Copyright (c) 2026 Sidero Labs, Inc.
//
// Use of this software is governed by the Business Source License
// included in the LICENSE file.

package config

// Features contains all Omni feature flags.
type Features struct {
	EnableTalosPreReleaseVersions bool `yaml:"enableTalosPreReleaseVersions"`
	EnableBreakGlassConfigs       bool `yaml:"enableBreakGlassConfigs"`
	EnableConfigDataCompression   bool `yaml:"enableConfigDataCompression"`
	EnableClusterImport           bool `yaml:"enableClusterImport"`

	DisableControllerRuntimeCache bool `yaml:"disableControllerRuntimeCache"`
}
